src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -395,6 +535,114 @@ private static DataGroup convertStarTableToDataGroup(StarTable table,
         return dataGroup;
     }
 
+
+    //This function is based on the packageValue function from the FitsStarTable class in the uk.ac.starlink.fits package
+    private static Object getValAsObject(Object base, int icol, boolean[] hasBlank, long[] blanks, boolean[] isScaled,
+                                         double[] scales, double[] zeros) {
+        if (base == null) {
```

Comment:
This function plus `getClassType` is pretty much the bulk of FitsStarTable.java.  I am not sure how this is an improvement unless we decided to abandon starlink completely.  But, if this is the decision, then I suggest writing more unit tests to ensure the commonly encountered tables are read in correctly.